<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>602</width>
    <height>737</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>601</width>
      <height>691</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Slot config</string>
     </attribute>
     <widget class="QPushButton" name="resetButton">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>540</y>
        <width>121</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Reload configuration</string>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>70</y>
        <width>571</width>
        <height>171</height>
       </rect>
      </property>
      <property name="title">
       <string>Secret key</string>
      </property>
      <widget class="QRadioButton" name="hexRadioButton">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>70</y>
         <width>82</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Hex</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QRadioButton" name="base32RadioButton">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>70</y>
         <width>82</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Base 32</string>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QLineEdit" name="secretEdit">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>30</y>
         <width>481</width>
         <height>20</height>
        </rect>
       </property>
       <property name="inputMask">
        <string>HH HH HH HH HH HH HH HH HH HH HH HH HH HH HH HH HH HH HH HH; </string>
       </property>
       <property name="text">
        <string>00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00</string>
       </property>
       <property name="maxLength">
        <number>59</number>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>30</y>
         <width>61</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Secret key:</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>70</y>
         <width>71</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Input format:</string>
       </property>
      </widget>
      <widget class="QCheckBox" name="enterCheckBox">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>100</y>
         <width>191</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Send 'enter'as the last keystroke</string>
       </property>
      </widget>
     </widget>
     <widget class="QPushButton" name="cancelButton">
      <property name="geometry">
       <rect>
        <x>450</x>
        <y>540</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Cancel</string>
      </property>
     </widget>
     <widget class="QPushButton" name="writeButton">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>540</y>
        <width>111</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Write configuration</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>270</x>
        <y>10</y>
        <width>71</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Define name:</string>
      </property>
     </widget>
     <widget class="QComboBox" name="slotComboBox">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>10</y>
        <width>191</width>
        <height>22</height>
       </rect>
      </property>
      <item>
       <property name="text">
        <string>HOTP slot 1</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>HOTP slot 2</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>TOTP slot 1</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>TOTP slot 2</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>TOTP slot 3</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>TOTP slot 4</string>
       </property>
      </item>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>10</y>
        <width>31</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Slot:</string>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>400</y>
        <width>571</width>
        <height>111</height>
       </rect>
      </property>
      <property name="title">
       <string>Token ID</string>
      </property>
      <widget class="QCheckBox" name="tokenIDCheckBox">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>101</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Send token ID</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="ompEdit">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>60</y>
         <width>31</width>
         <height>20</height>
        </rect>
       </property>
       <property name="inputMask">
        <string>NN; </string>
       </property>
      </widget>
      <widget class="QLineEdit" name="ttEdit">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>60</y>
         <width>31</width>
         <height>20</height>
        </rect>
       </property>
       <property name="inputMask">
        <string>NN; </string>
       </property>
      </widget>
      <widget class="QLineEdit" name="muiEdit">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>60</y>
         <width>91</width>
         <height>20</height>
        </rect>
       </property>
       <property name="inputMask">
        <string>HHHHHHHH; </string>
       </property>
      </widget>
      <widget class="QLabel" name="label_7">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>31</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>OMP:</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_8">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>60</y>
         <width>21</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>TT:</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_9">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>60</y>
         <width>31</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>MUI:</string>
       </property>
      </widget>
      <widget class="QComboBox" name="keyboardComboBox">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>290</x>
         <y>60</y>
         <width>131</width>
         <height>22</height>
        </rect>
       </property>
       <item>
        <property name="text">
         <string>QWERTY</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>QWERTZ</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>AZERTY</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>DVORAK</string>
        </property>
       </item>
      </widget>
      <widget class="QLabel" name="label_10">
       <property name="geometry">
        <rect>
         <x>290</x>
         <y>30</y>
         <width>91</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Keyboard layout:</string>
       </property>
      </widget>
     </widget>
     <widget class="QLineEdit" name="nameEdit">
      <property name="geometry">
       <rect>
        <x>340</x>
        <y>10</y>
        <width>113</width>
        <height>20</height>
       </rect>
      </property>
      <property name="inputMask">
       <string>xxxxxxxxxxxxxxx; </string>
      </property>
     </widget>
     <widget class="QGroupBox" name="hotpGroupBox">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>250</y>
        <width>571</width>
        <height>141</height>
       </rect>
      </property>
      <property name="title">
       <string>OATH-HOTP Parameters</string>
      </property>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>30</y>
         <width>71</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>HOTP length:</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="digits6radioButton">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>30</y>
         <width>82</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>6 digits</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QRadioButton" name="digits8radioButton">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>30</y>
         <width>82</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>8 digits</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_6">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>60</y>
         <width>101</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Moving factor seed:</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="counterEdit">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>60</y>
         <width>241</width>
         <height>20</height>
        </rect>
       </property>
       <property name="inputMask">
        <string>hhhhhhhhhhhhhhhh; </string>
       </property>
       <property name="text">
        <string>0000000000000000</string>
       </property>
      </widget>
      <widget class="QPushButton" name="setToZeroButton">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>90</y>
         <width>75</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Set to zero</string>
       </property>
      </widget>
      <widget class="QPushButton" name="setToRandomButton">
       <property name="geometry">
        <rect>
         <x>220</x>
         <y>90</y>
         <width>91</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Set to random</string>
       </property>
      </widget>
     </widget>
     <widget class="QPushButton" name="eraseButton">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>620</y>
        <width>71</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Erase slot</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>General config</string>
     </attribute>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>20</y>
        <width>101</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Firmware version:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_12">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>50</y>
        <width>101</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Card serial number:</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="firmwareEdit">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>20</y>
        <width>113</width>
        <height>20</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLineEdit" name="serialEdit">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>50</y>
        <width>113</width>
        <height>20</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="label_13">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>90</y>
        <width>101</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>NumLock dobletap:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_14">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>120</y>
        <width>111</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>CapsLock doubletap:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_15">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>150</y>
        <width>111</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>ScrollLock doubletap:</string>
      </property>
     </widget>
     <widget class="QComboBox" name="numLockComboBox">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>90</y>
        <width>131</width>
        <height>22</height>
       </rect>
      </property>
      <item>
       <property name="text">
        <string>Do nothing</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Send HOTP1</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Send HOTP2</string>
       </property>
      </item>
     </widget>
     <widget class="QComboBox" name="capsLockComboBox">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>120</y>
        <width>131</width>
        <height>22</height>
       </rect>
      </property>
      <item>
       <property name="text">
        <string>Do nothing</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Send HOTP1</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Send HOTP2</string>
       </property>
      </item>
     </widget>
     <widget class="QComboBox" name="scrollLockComboBox">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>150</y>
        <width>131</width>
        <height>22</height>
       </rect>
      </property>
      <item>
       <property name="text">
        <string>Do nothing</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Send HOTP1</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Send HOTP2</string>
       </property>
      </item>
     </widget>
     <widget class="QPushButton" name="writeGeneralConfigButton">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>200</y>
        <width>101</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Write config</string>
      </property>
     </widget>
     <widget class="QPushButton" name="resetGeneralConfigButton">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>200</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Reset</string>
      </property>
     </widget>
     <widget class="QPushButton" name="generalCancelButton">
      <property name="geometry">
       <rect>
        <x>270</x>
        <y>200</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Cancel</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>602</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>cancelButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>hide()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>267</x>
     <y>692</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>366</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>generalCancelButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>hide()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>309</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>300</x>
     <y>368</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
